extensions/web/08_text_encoding.js

```diff
@@ -4338,6 +4338,129 @@
     }
   }
 
+  class TextDecoderStream {
+    #decoder;
+    #transform;
+
+    get encoding() {
+      return this.#decoder.encoding;
+    }
+    get fatal() {
+      return this.#decoder.fatal;
+    }
+    get ignoreBOM() {
+      return this.#decoder.ignoreBOM;
+    }
+
+    constructor(label = "utf-8", options = { fatal: false }) {
+      this.#decoder = new TextDecoder(label, options);
+      this.#transform = new TransformStream({
+        // The transform and flush functions need access to TextDecoderStream's
+        // `this`, so they are defined as functions rather than methods.
+        transform: (chunk, controller) => {
+          try {
+            // Step 1 in "decode and enqueue a chunk" converts `chunk` to an
+            // `[AllowShared] BufferSource`, which is the same type as the first
+            // argument to TextDecoder.prototype.decode – except that that
+            // argument is marked as optional. Therefore, we need to
+            // special-case `undefined`.
+            if (chunk === undefined) {
+              throw new TypeError(
+                "Chunk in the stream is not of type ArrayBuffer or ArrayBufferView",
+              );
+            }
+
+            const decoded = this.#decoder.decode(chunk, { stream: true });
+            if (decoded) {
+              controller.enqueue(decoded);
+            }
+            return Promise.resolve();
+          } catch (err) {
+            return Promise.reject(err);
+          }
+        },
+        flush: (controller) => {
+          try {
+            const final = this.#decoder.decode();
+            if (final) {
+              controller.enqueue(final);
+            }
+            return Promise.resolve();
+          } catch (err) {
+            return Promise.reject(err);
+          }
+        },
+      });
+    }
+
+    get readable() {
+      return this.#transform.readable;
+    }
+    get writable() {
+      return this.#transform.writable;
+    }
+
+    get [Symbol.toStringTag]() {
+      return "TextDecoderStream";
+    }
+  }
+
+  class TextEncoderStream {
+    #pendingHighSurrogate = null;
+    #transform;
+
+    encoding = "utf-8";
```

Comment:
This should be stored as internal state (private field or symbol field), and then expose to user as a getter. The getter can always return "utf-8". See https://github.com/denoland/deno/blob/main/extensions/web/08_text_encoding.js#L146-L150